README.rst
Python Client for Cloud Vision API
==================================

|stable| |pypi| |versions|

`Cloud Vision API`_: 

- `Client Library Documentation`_
- `Product Documentation`_

.. |stable| image:: https://img.shields.io/badge/support-stable-gold.svg
   :target: https://github.com/googleapis/google-cloud-python/blob/main/README.rst#stability-levels
.. |pypi| image:: https://img.shields.io/pypi/v/google-cloud-vision.svg
   :target: https://pypi.org/project/google-cloud-vision/
.. |versions| image:: https://img.shields.io/pypi/pyversions/google-cloud-vision.svg
   :target: https://pypi.org/project/google-cloud-vision/
.. _Cloud Vision API: https://cloud.google.com/vision/docs/
.. _Client Library Documentation: https://cloud.google.com/python/docs/reference/vision/latest
.. _Product Documentation:  https://cloud.google.com/vision/docs/

Quick Start
-----------

In order to use this library, you first need to go through the following steps:

1. `Select or create a Cloud Platform project.`_
2. `Enable billing for your project.`_
3. `Enable the Cloud Vision API.`_
4. `Setup Authentication.`_

.. _Select or create a Cloud Platform project.: https://console.cloud.google.com/project
.. _Enable billing for your project.: https://cloud.google.com/billing/docs/how-to/modify-project#enable_billing_for_a_project
.. _Enable the Cloud Vision API.:  https://cloud.google.com/vision/docs/
.. _Setup Authentication.: https://googleapis.dev/python/google-api-core/latest/auth.html

Installation
~~~~~~~~~~~~

Install this library in a `virtualenv`_ using pip. `virtualenv`_ is a tool to
create isolated Python environments. The basic problem it addresses is one of
dependencies and versions, and indirectly permissions.

With `virtualenv`_, it's possible to install this library without needing system
install permissions, and without clashing with the installed system
dependencies.

.. _`virtualenv`: https://virtualenv.pypa.io/en/latest/


Code samples and snippets
~~~~~~~~~~~~~~~~~~~~~~~~~

Code samples and snippets live in the `samples/` folder.


Supported Python Versions
^^^^^^^^^^^^^^^^^^^^^^^^^
Our client libraries are compatible with all current [active](https://devguide.python.org/devcycle/#in-development-main-branch) and [maintenance](https://devguide.python.org/devcycle/#maintenance-branches) versions of
Python.

Python >= 3.7

Unsupported Python Versions
^^^^^^^^^^^^^^^^^^^^^^^^^^^
Python <= 3.6

If you are using an [end-of-life](https://devguide.python.org/devcycle/#end-of-life-branches)
version of Python, we recommend that you update as soon as possible to an actively supported version.


Mac/Linux
^^^^^^^^^

.. code-block:: console

    pip install virtualenv
    virtualenv <your-env>
    source <your-env>/bin/activate
    <your-env>/bin/pip install google-cloud-vision


Windows
^^^^^^^

.. code-block:: console

    pip install virtualenv
    virtualenv <your-env>
    <your-env>\Scripts\activate
    <your-env>\Scripts\pip.exe install google-cloud-vision

Next Steps
~~~~~~~~~~

-  Read the `Client Library Documentation`_ for Cloud Vision API
   to see other available methods on the client.
-  Read the `Cloud Vision API Product documentation`_ to learn
   more about the product and see How-to Guides.
-  View this `README`_ to see the full list of Cloud
   APIs that we cover.

.. _Cloud Vision API Product documentation:  https://cloud.google.com/vision/docs/
.. _README: https://github.com/googleapis/google-cloud-python/blob/main/README.rst


CHANGELOG.md
# Changelog

[PyPI History][1]

[1]: https://pypi.org/project/google-cloud-vision/#history

## [1.0.2](https://github.com/googleapis/python-vision/compare/v1.0.1...v1.0.2) (2022-06-09)


### Bug Fixes

* **deps:** require protobuf<4.0.0 on v1 branch ([#445](https://github.com/googleapis/python-vision/issues/445)) ([9623c1f](https://github.com/googleapis/python-vision/commit/9623c1f5d9a161c54f552910c10e8fb2b7c0b2ca))

### [1.0.1](https://github.com/googleapis/python-vision/compare/v1.0.0...v1.0.1) (2022-04-01)


### Bug Fixes

* **deps:** require google-api-core >= 1.31.5, >= 2.3.2 on v1 release ([#324](https://github.com/googleapis/python-vision/issues/324)) ([a3d3d66](https://github.com/googleapis/python-vision/commit/a3d3d665b7982598eb816c8fcfb3a66078c93a06))

## [1.0.0](https://www.github.com/googleapis/python-vision/compare/v0.42.0...v1.0.0) (2020-02-28)


### Features

* bump release status to GA ([#11](https://www.github.com/googleapis/python-vision/issues/11)) ([2129bde](https://www.github.com/googleapis/python-vision/commit/2129bdedfa0dca85c5adc5350bff10d4a485df77))

## [0.42.0](https://www.github.com/googleapis/python-vision/compare/v0.41.0...v0.42.0) (2020-02-03)


### Features

* **vision:** undeprecate resource name helpers, add 2.7 sunset warning (via synth) ([#10052](https://www.github.com/googleapis/python-vision/issues/10052)) ([2c86705](https://www.github.com/googleapis/python-vision/commit/2c86705154ce219f8fb6bbbcc80832ec2b63baeb))

## 0.41.0

12-05-2019 14:49 PST


### New Features
- Add deprecation warning to resource name helper functions. ([#9866](https://github.com/googleapis/google-cloud-python/pull/9866))

### Documentation
- Add deprecation information to resource name helper function documentation. ([#9866](https://github.com/googleapis/google-cloud-python/pull/9866))

## 0.40.0

11-08-2019 09:30 PST


### Implementation Changes
- Add proto annotations (via synth). ([#9441](https://github.com/GoogleCloudPlatform/google-cloud-python/pull/9441))
- Add RPC annotations, update docstrings (via synth). ([#9230](https://github.com/GoogleCloudPlatform/google-cloud-python/pull/9230))

### New Features
- Add celebrity recognition support to v1p4beta1 (via synth). ([#9613](https://github.com/GoogleCloudPlatform/google-cloud-python/pull/9613))
- Add object annotation support to v1p4beta1 (via synth). ([#9613](https://github.com/GoogleCloudPlatform/google-cloud-python/pull/9613))
- Add `purge_products` method to v1p4beta1 (via synth). ([#9613](https://github.com/GoogleCloudPlatform/google-cloud-python/pull/9613))
- Add more product categories (via synth). ([#9224](https://github.com/GoogleCloudPlatform/google-cloud-python/pull/9224))

### Documentation
- Update docstring for `product_category`, change requests intersphinx url (via synth). ([#9413](https://github.com/GoogleCloudPlatform/google-cloud-python/pull/9413))
- Fix intersphinx reference to requests. ([#9294](https://github.com/GoogleCloudPlatform/google-cloud-python/pull/9294))
- Remove compatability badges from READMEs. ([#9035](https://github.com/GoogleCloudPlatform/google-cloud-python/pull/9035))
- Update docstring to say at most 5 frames can be batch annotated (via synth). ([#9041](https://github.com/GoogleCloudPlatform/google-cloud-python/pull/9041))

### Internal / Testing Changes
- Flatten case of logo text in systest. ([#9159](https://github.com/GoogleCloudPlatform/google-cloud-python/pull/9159))
- Docs: Remove CI for gh-pages, use googleapis.dev for `api_core` refs. ([#9085](https://github.com/GoogleCloudPlatform/google-cloud-python/pull/9085))

## 0.39.0

08-12-2019 13:57 PDT

### Implementation Changes
- Remove send/recv msg size limit (via synth). ([#8976](https://github.com/googleapis/google-cloud-python/pull/8976))

### New Features
- Add 'parent' argument to annotation requests; add 'ProductSearchClient.purge_products' method; add 'object_annotations' field to product search results (via synth). ([#8988](https://github.com/googleapis/google-cloud-python/pull/8988))

### Documentation
- Update intersphinx mapping for requests. ([#8805](https://github.com/googleapis/google-cloud-python/pull/8805))

## 0.38.1

07-24-2019 17:54 PDT

### Dependencies
- Bump minimum version for google-api-core to 1.14.0. ([#8709](https://github.com/googleapis/google-cloud-python/pull/8709))

### Documentation
- Link to googleapis.dev documentation in READMEs. ([#8705](https://github.com/googleapis/google-cloud-python/pull/8705))
- Add compatibility check badges to READMEs. ([#8288](https://github.com/googleapis/google-cloud-python/pull/8288))

## 0.38.0

07-09-2019 10:05 PDT


### Implementation Changes
- Allow kwargs to be passed to create_channel (via synth). ([#8411](https://github.com/googleapis/google-cloud-python/pull/8411))

### New Features
- Add 'client_options' support, update list method docstrings (via synth). ([#8529](https://github.com/googleapis/google-cloud-python/pull/8529))

### Internal / Testing Changes
- Pin black version (via synth). ([#8602](https://github.com/googleapis/google-cloud-python/pull/8602))
- Add docs job to publish to googleapis.dev. ([#8464](https://github.com/googleapis/google-cloud-python/pull/8464))
- All of the negative vision vpc sc tests should use the same file. ([#8439](https://github.com/googleapis/google-cloud-python/pull/8439))
- In order to make sure access to the bucket is blocked by vpc sc, try to download any arbitrary file from the bucket. ([#8374](https://github.com/googleapis/google-cloud-python/pull/8374))
- Add disclaimer to auto-generated template files (via synth). ([#8335](https://github.com/googleapis/google-cloud-python/pull/8335))
- Add Vision API system tests to verify accessing a gcs bucket outside of a secure perimeter is blocked. ([#8276](https://github.com/googleapis/google-cloud-python/pull/8276))

owl-bot-staging/v1p2beta1/noxfile.py
# -*- coding: utf-8 -*-
# Copyright 2022 Google LLC
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
#
import os
import pathlib
import shutil
import subprocess
import sys


import nox  # type: ignore

ALL_PYTHON = [
    "3.6",
    "3.7",
    "3.8",
    "3.9",
    "3.10",
]

CURRENT_DIRECTORY = pathlib.Path(__file__).parent.absolute()

LOWER_BOUND_CONSTRAINTS_FILE = CURRENT_DIRECTORY / "constraints.txt"
PACKAGE_NAME = subprocess.check_output([sys.executable, "setup.py", "--name"], encoding="utf-8")

BLACK_VERSION = "black==19.10b0"
BLACK_PATHS = ["docs", "google", "tests", "samples", "noxfile.py", "setup.py"]
DEFAULT_PYTHON_VERSION = "3.9"

nox.sessions = [
    "unit",
    "cover",
    "mypy",
    "check_lower_bounds"
    # exclude update_lower_bounds from default
    "docs",
    "blacken",
    "lint",
    "lint_setup_py",
]

@nox.session(python=ALL_PYTHON)
def unit(session):
    """Run the unit test suite."""

    session.install('coverage', 'pytest', 'pytest-cov', 'asyncmock', 'pytest-asyncio')
    session.install('-e', '.')

    session.run(
        'py.test',
        '--quiet',
        '--cov=google/cloud/vision_v1p2beta1/',
        '--cov=tests/',
        '--cov-config=.coveragerc',
        '--cov-report=term',
        '--cov-report=html',
        os.path.join('tests', 'unit', ''.join(session.posargs))
    )


@nox.session(python=DEFAULT_PYTHON_VERSION)
def cover(session):
    """Run the final coverage report.
    This outputs the coverage report aggregating coverage from the unit
    test runs (not system test runs), and then erases coverage data.
    """
    session.install("coverage", "pytest-cov")
    session.run("coverage", "report", "--show-missing", "--fail-under=100")

    session.run("coverage", "erase")


@nox.session(python=ALL_PYTHON)
def mypy(session):
    """Run the type checker."""
    session.install('mypy', 'types-pkg_resources')
    session.install('.')
    session.run(
        'mypy',
        '--explicit-package-bases',
        'google',
    )


@nox.session
def update_lower_bounds(session):
    """Update lower bounds in constraints.txt to match setup.py"""
    session.install('google-cloud-testutils')
    session.install('.')

    session.run(
        'lower-bound-checker',
        'update',
        '--package-name',
        PACKAGE_NAME,
        '--constraints-file',
        str(LOWER_BOUND_CONSTRAINTS_FILE),
    )


@nox.session
def check_lower_bounds(session):
    """Check lower bounds in setup.py are reflected in constraints file"""
    session.install('google-cloud-testutils')
    session.install('.')

    session.run(
        'lower-bound-checker',
        'check',
        '--package-name',
        PACKAGE_NAME,
        '--constraints-file',
        str(LOWER_BOUND_CONSTRAINTS_FILE),
    )

@nox.session(python=DEFAULT_PYTHON_VERSION)
def docs(session):
    """Build the docs for this library."""

    session.install("-e", ".")
    session.install("sphinx==4.0.1", "alabaster", "recommonmark")

    shutil.rmtree(os.path.join("docs", "_build"), ignore_errors=True)
    session.run(
        "sphinx-build",
        "-W",  # warnings as errors
        "-T",  # show full traceback on exception
        "-N",  # no colors
        "-b",
        "html",
        "-d",
        os.path.join("docs", "_build", "doctrees", ""),
        os.path.join("docs", ""),
        os.path.join("docs", "_build", "html", ""),
    )


@nox.session(python=DEFAULT_PYTHON_VERSION)
def lint(session):
    """Run linters.

    Returns a failure if the linters find linting errors or sufficiently
    serious code quality issues.
    """
    session.install("flake8", BLACK_VERSION)
    session.run(
        "black",
        "--check",
        *BLACK_PATHS,
    )
    session.run("flake8", "google", "tests", "samples")


@nox.session(python=DEFAULT_PYTHON_VERSION)
def blacken(session):
    """Run black. Format code to uniform standard."""
    session.install(BLACK_VERSION)
    session.run(
        "black",
        *BLACK_PATHS,
    )


@nox.session(python=DEFAULT_PYTHON_VERSION)
def lint_setup_py(session):
    """Verify that setup.py is valid (including RST check)."""
    session.install("docutils", "pygments")
    session.run("python", "setup.py", "check", "--restructuredtext", "--strict")
